File: problems/888-fair-candy-swap/solution.py
class Solution:
    def fairCandySwap(self, aliceSizes: List[int], bobSizes: List[int]) -> List[int]:
        sum_alice = sum(aliceSizes)
        sum_bob = sum(bobSizes)
        set_bob = set(bobSizes)
        for x in aliceSizes:
            y = (sum_bob - sum_alice + 2 * x) // 2
            if y in set_bob:
                return [x, y]

File: problems/888-fair-candy-swap/solution.java
class Solution {
    public int[] fairCandySwap(int[] aliceSizes, int[] bobSizes) {
        int sumAlice = 0;
        int sumBob = 0;
        for (int size : aliceSizes) {
            sumAlice += size;
        }
        for (int size : bobSizes) {
            sumBob += size;
        }
        Set<Integer> setBob = new HashSet<>();
        for (int size : bobSizes) {
            setBob.add(size);
        }
        for (int size : aliceSizes) {
            int target = (sumAlice - size + sumBob - size) / 2;
            if (setBob.contains(target)) {
                return new int[]{size, target};
            }
        }
        return new int[]{};
    }
}